mesh/ballotwriter/ballotwriter_test.go

```diff
@@ -0,0 +1,223 @@
+package ballotwriter_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zaptest"
+
+	"github.com/spacemeshos/go-spacemesh/codec"
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/malfeasance/wire"
+	"github.com/spacemeshos/go-spacemesh/mesh/ballotwriter"
+	"github.com/spacemeshos/go-spacemesh/signing"
+	"github.com/spacemeshos/go-spacemesh/sql"
+	"github.com/spacemeshos/go-spacemesh/sql/ballots"
+	"github.com/spacemeshos/go-spacemesh/sql/identities"
+	"github.com/spacemeshos/go-spacemesh/sql/statesql"
+	"github.com/spacemeshos/go-spacemesh/sql/statesql/migrations"
+)
+
+func init() {
+	// otherwise we get a divide by zero error
+	types.SetLayersPerEpoch(1)
+}
```

Comment:
It would be better to put that into `TestMain` otherwise this might pollute the global state of the application.